readme.md

```diff
@@ -1,60 +1,148 @@
+```bash
+-e TZ="America/Chicago"
 ```
+
+### Add your custom keys
+
+To use your own keys pass as arguments on startup (post 1.0.2 pre 1.0.2 must manually pass as shown below)
+```bash
 -v /path/to/key/file:/path/in/container -e keys="/path/in/container"
 ```
 or pass keys into the running container
-```
+```bash
 docker exec -it <container-name> venv/bin/chia keys add
 ```
 alternatively you can pass in your local keychain, if you have previously deployed chia with these keys on the host machine
-```
+```bash
 -v ~/.local/share/python_keyring/:/root/.local/share/python_keyring/
 ```
 
-To start a farmer only node pass
+### Persist configuration and db
+
+You can persist whole db and configuration, simply mount it to Host.
+```bash
+-v ~/.chia:/root/.chia
 ```
+
+### Farmer only
+
+To start a farmer only node pass
+```bash
 -e farmer="true"
 ```
 
+### Harverster only
+
 To start a harvester only node pass
-```
+```bash
 -e harvester="true" -e farmer_address="addres.of.farmer" -e farmer_port="portnumber" -v /path/to/ssl/ca:/path/in/container -e ca="/path/in/container" -e keys="copy"
 ```
 
+### Plots
+
 The `plots_dir` environment variable can be used to specify the directory containing the plots, it supports PATH-style colon-separated directories.
 
-#### or run commands externally with venv (this works for most chia XYZ commands)
+Or, you can cimply mount `/plots` path to your host machine.
+
+### Docker Compose
+
+```yaml
+version: "3.6"
+services:
+  chia:
+    container_name: chia
+    restart: always
+    image: ghcr.io/chia-network/chia:latest
+    ports:
+      - 8447:8447
+    environment:
+      # Farmer Only    
+#     - farmer="true"
+      # Harvester Only
+#     - harvester="true"
+#     - farmer_address="192.168.0.10" 
+#     - farmer_port="8447"
+#     - ca="/path/in/container"
+#     - keys="copy"
+      # Harvester Only END
+      # If you would like to add keys manually via mnemonic file
+#     - keys="/path/in/container"
+      # OR
+      # Disable key generation on start
+#     - keys=""
+      - TZ=${TZ}
+    volumes:
+      - /path/to/plots:/plots
+      - /home/user/.chia:/root/.chia
+#     - /home/user/mnemonic:/path/in/container
 ```
+
+## CLI
+
+You can run commands externally with venv (this works for most chia [CLI commands](https://github.com/Chia-Network/chia-blockchain/wiki/CLI-Commands-Reference))
+```bash
 docker exec -it chia venv/bin/chia plots add -d /plots
 ```
 
-#### status from outside the container
-```
+### Is it working?
+
+You can see status from outside the container
+```bash
 docker exec -it chia venv/bin/chia show -s -c
 ```
-
-#### Connect to testnet?
+or
+```bash
+docker exec -it chia venv/bin/chia farm summary
 ```
+
+### Connect to testnet?
+
+```bash
 docker run -d --expose=58444 --expose=8555 -e testnet=true --name <container-name> ghcr.io/chia-network/chia:latest
```

Comment:
```suggestion
docker run -d --expose=58444 -e testnet=true --name chia ghcr.io/chia-network/chia:latest
```

Reply:
What was the purpose of 8555?

Reply:
8555 is full node RPC - so there may be a reason to open it if someone is doing things with RPC from outside the container, but probably not in the default examples